pkg/controllers/user/pipeline/controller/clusterpipeline/clusterpipelinesyncer.go

```diff
@@ -0,0 +1,222 @@
+package clusterpipeline
+
+import (
+	"github.com/pkg/errors"
+	"github.com/rancher/rancher/pkg/controllers/user/pipeline/utils"
+	"github.com/rancher/types/apis/apps/v1beta2"
+	"github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	coretypev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	rbactypev1 "k8s.io/client-go/kubernetes/typed/rbac/v1"
+)
+
+type Syncer struct {
+	namespaces          v1.NamespaceInterface
```

Comment:
@gitlawr We should really use listers and fetch data from cache unless create/update/delete is needed. Can you revise all the fields below to check if they can be replaced  by listers.

Reply:
Or if the field is used for all list/delete/update, consider adding a Lister in addition to the interface, to the list of fields.